base/src/main/java/com/smartdevicelink/managers/screen/SoftButtonObject.java

```diff
@@ -264,6 +266,13 @@ public void setName(@NonNull String name) {
      * @param states a list of the object's soft button states. <strong>states should be unique for every SoftButtonObject. A SoftButtonState instance cannot be reused for multiple SoftButtonObjects.</strong>
      */
     public void setStates(@NonNull List<SoftButtonState> states) {
+        // If the list of states is empty, throw an error with DebugTool and return
+        attemptedToAssignEmptyStateList = states.isEmpty();
+        if (attemptedToAssignEmptyStateList) {
+            DebugTool.logError(TAG,"The state list is empty");
+            return;
+        }
```

Comment:
Since we are doing it the constructor i think it also makes sense to check the list does not have duplicate state names here as well.
```
if (hasTwoStatesOfSameName(states)) {
    //Log error
    //return
}
```

Reply:
That sounds reasonable to me, I just wasn't sure I should increase the scope of this PR

Reply:
After looking at the iOS implementation the iOS team will also need to make changes. With the teams input I think it makes sense to add this check to setStates

Reply:
See my above comment